Branch-SDK/src/io/branch/referral/Branch.java

```diff
@@ -393,7 +393,11 @@ public static Branch getInstance(Context context, String branchKey) {
 		}
 		branchReferral_.context_ = context.getApplicationContext();
 		if (branchKey.startsWith("key_")) {
-			branchReferral_.prefHelper_.setBranchKey(branchKey);
+			boolean isNewBranchKeySet = branchReferral_.prefHelper_.setBranchKey(branchKey);
+			//on setting a new key clear link cache
+			if (isNewBranchKeySet) {
+				branchReferral_.linkCache_.clear();
```

Comment:
Instead, the goal of "clearing when key changes" is supposed to remove identity, install/session params, etc.